### ISSUE TITLE
Kind: provide an option to pass a configuration yaml to the kind

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#2215]
### Is your enhancement related to a problem? Please describe

It would be nice to a have way to pass a [configuration] to the kind cluster in a form of yaml file.

[configuration]: https://kind.sigs.k8s.io/docs/user/configuration/

### Describe the solution you'd like

Provide an option to pass a configuration file to the kind cluster creation page. I believe there should be a switch in order to avoid conflicts between port mapping, ingress setup and other parameters passed by a form.

**Comment by odockal** [comment]
@cdrage Can you please help me here? Am I mistaken if I think that this issue is basically implemented by a `Apply YAML` feature available in Kubernetes Pages?

**Comment by cdrage** [comment]
> @cdrage Can you please help me here? Am I mistaken if I think that this issue is basically implemented by a `Apply YAML` feature available in Kubernetes Pages?

Yup, you can apply any YAML now, and thus you can apply it to your kind cluster.

**Comment by cmoulliard** [comment]
> Can you please help me here? Am I mistaken if I think that this issue is basically implemented by a `Apply YAML` feature available in Kubernetes Pages?

No as the kind config file should be passed to kind when the cluster is created. In a terminal this is what you will do
```
kind create cluster --config <path to a kind config file>
```

Such a config file should be uploaded here (to override the default parameters) 
<img width="1060" alt="Screenshot 2024-04-18 at 11 40 13" src="https://github.com/containers/podman-desktop/assets/463790/fb78be72-6cf8-4ee1-8773-f10fe65abffa">


See kind doc: https://kind.sigs.k8s.io/docs/user/configuration/#getting-started

**Comment by cmoulliard** [comment]
This ticket should be re-opened !!

**Comment by odockal** [comment]
@cmoulliard Reopened. Thank for inputs.
@cdrage I think this is valid point, since one can pass config during cluster creation which cannot be covered by post action. Podman Desktop could offer that.

**Comment by jeffmaury** [comment]
Still nice to have

**Comment by odockal** [comment]
@jeffmaury I will add it into next sprint.

**Comment by gastoner** [comment]
I have two ideas how to do this:
1. Add a switch button that would switch between configFile or form configuration 
2. Add a reset button to file input, if will be the file passed

If would be the switch turned on/off, or if the file is passed the other options could be either not visible, or disabled(gray)
cc @ekidneyrh @MariaLeonova @slemeur

**Comment by slemeur** [comment]
I think that's a great idea. 

Another way, is that the first field could be to optionaly provide the configuration yaml file. A bit like we do on the build container. If that field has a valid value, the other options might reflect what's in the file and would be disabled (gray)

<img width="1587" alt="Image" src="https://github.com/user-attachments/assets/91abf873-2bc1-4e8b-9455-cec876b76b36" />

**Comment by gastoner** [comment]
> I think that's a great idea.
> 
> Another way, is that the first field could be to optionaly provide the configuration yaml file. A bit like we do on the build container. If that field has a valid value, the other options might reflect what's in the file and would be disabled (gray)
> <img alt="Image" width="1587" src="https://private-user-images.githubusercontent.com/1636769/403474514-91abf873-2bc1-4e8b-9455-cec876b76b36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5NTQ3OTYsIm5iZiI6MTczNjk1NDQ5NiwicGF0aCI6Ii8xNjM2NzY5LzQwMzQ3NDUxNC05MWFiZjg3My0yYmMxLTRlOGItOTQ1NS1jZWM4NzZiNzZiMzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTVUMTUyMTM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTY5M2YzMzVlMTAyYjJjZjVhMzU4YWJlNjg3ZmYxOWYzYWQ5ZDMzNTc5OWUxMjM2MWZhZmYxNzYxOGNkZTc2YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.T2zqLwNpXOb5DA7Ck5QGrBMUBYBmKuli_C2FgN0kb-Q">

Yess, that is also an option, I'm just not sure if we want to have file above the cluster name. It is probably not a big thing but I would expect the name as a first record

**Comment by jeffmaury** [comment]
I would propose a 2 step approach:

1. Add optional config file
2. Enhance with form support

2 would be based on telemetry feedback
